nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/controller/StandardProcessorNode.java

```diff
@@ -1090,6 +1090,16 @@ public int getActiveThreadCount() {
                     break;
                 }
             }
+
+            // Ensure that execution node will not be misused
+            if (getExecutionNode() == ExecutionNode.PRIMARY) {
+                if (getInputRequirement() == Requirement.INPUT_REQUIRED
+                        || getInputRequirement() == Requirement.INPUT_ALLOWED && !getIncomingConnections().isEmpty()) {
+                    results.add(new ValidationResult.Builder()
+                            .explanation("Processor should set to run on all nodes.")
```

Comment:
Probably need to provide more details in the explanation. Why should the processor be set to run on all nodes? Perhaps something to the effect of "Processors with incoming connections cannot be scheduled for Primary Node Only."